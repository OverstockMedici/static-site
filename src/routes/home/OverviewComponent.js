import React, { Component } from 'react';

const OverviewComponent = ({ styleName, heading, content }) => (
    <div data-testid="overview" className={styleName}>
        <h2>{heading}</h2>
        <p>{content}</p>
    </div>
);

export default OverviewComponent;
