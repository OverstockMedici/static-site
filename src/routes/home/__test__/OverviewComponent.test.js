import React from 'react';
import ReactDOM from 'react-dom';
import OverviewComponent from './../OverviewComponent';

import { render, cleanup } from '@testing-library/react';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<OverviewComponent />, div);
});

it('renders OverviewComponent correctly', () => {
    const { getByTestId, container } = render(
        <OverviewComponent
            heading="Democratize Capital"
            content="lorem ipsum"
            styleName="overview-1"
        />,
    );
    expect(getByTestId('overview')).toHaveTextContent('Democratize Capital');
    expect(getByTestId('overview')).toHaveTextContent('lorem ipsum');
    expect(container.firstChild).toHaveClass('overview-1');
});

it('matches snapshot', () => {
    const overviewTree = render(
        <OverviewComponent
            heading="Democratize Capital"
            content="lorem ipsum"
            styleName="overview-1"
        />,
    );

    expect(overviewTree.firstChild).toMatchSnapshot();
});
