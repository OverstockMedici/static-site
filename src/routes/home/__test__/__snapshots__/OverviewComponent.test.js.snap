// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="overview-1"
  data-testid="overview"
>
  <h2>
    Democratize Capital
  </h2>
  <p>
    lorem ipsum
  </p>
</div>
`;
